recipes/libmodbus/all/conanfile.py

```diff
@@ -39,85 +41,109 @@ def config_options(self):
+            env.define("AR", f"{ar_lib} \"lib -nologo -verbose\"")
+            env.define("RANLIB", ":")
+            env.define("STRING", ":")
+            env.define("NM", "dumpbin -symbols")
+
+        tc.generate(env)
+
+        # generate dependencies for pkg-config
+        tc = PkgConfigDeps(self)
+        tc.generate()
+
+        # generate dependencies for autotools
+        tc = AutotoolsDeps(self)
+        tc.generate()
+
+        # inject tools_require env vars in build context
+        ms = VirtualBuildEnv(self)
+        ms.generate()
 
     def _patch_sources(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
+        apply_conandata_patches(self)
         if not self.options.shared:
             for decl in ("__declspec(dllexport)", "__declspec(dllimport)"):
-                tools.replace_in_file(os.path.join(self._source_subfolder, "src", "modbus.h"), decl, "")
+                replace_in_file(self, os.path.join(self.source_folder, "src", "modbus.h"), decl, "")
 
     def build(self):
         self._patch_sources()
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.make()
+        autotools = Autotools(self)
+        autotools.configure()
+        autotools.make()
 
     def package(self):
-        self.copy("COPYING*", src=self._source_subfolder, dst="licenses")
-        with self._build_context():
-            autotools = self._configure_autotools()
-            autotools.install()
-
-        os.unlink(os.path.join(self.package_folder, "lib", "libmodbus.la"))
-        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
-        tools.rmdir(os.path.join(self.package_folder, "share"))
+        copy(self, pattern="COPYING*", dst=os.path.join(self.package_folder, "licenses"), src=self.source_folder)
+        autotools = Autotools(self)
+        # see https://github.com/conan-io/conan/issues/12006
+        autotools.install(args=[f"DESTDIR={unix_path(self, self.package_folder)}"])
+
+        if is_apple_os(self):
+            fix_apple_shared_install_name(self)
+
+        rm(self, "*.la", os.path.join(self.package_folder, "lib"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "share"))
         if self.settings.compiler == "Visual Studio" and self.options.shared:
-            tools.rename(os.path.join(self.package_folder, "lib", "modbus.dll.lib"),
-                         os.path.join(self.package_folder, "lib", "modbus.lib"))
+            rename(self,
+                    os.path.join(self.package_folder, "lib", "modbus.dll.lib"),
+                    os.path.join(self.package_folder, "lib", "modbus.lib"))
 
     def package_info(self):
-        self.cpp_info.names["pkg_config"] = "libmodbus"
-        self.cpp_info.includedirs.append(os.path.join("include", "modbus"))
```

Comment:
Why is it removed? According to pkgconfig file installed by libmodbus, it should not be removed.

Reply:
It was commented out and later removed in commit 379bf76d5ff3c5fd93009984aad71a4c331c0758 since the test package wasn't relying on it.
But you're totally right.